.github/workflows/unit_tests.yml

```diff
@@ -144,12 +145,12 @@ jobs:
       uses: actions/cache@v3
       with:
         path: ${{ env.pythonLocation }}
-        key: test-0-${{ env.pythonLocation }}-${{ hashFiles('setup.cfg') }}-${{ hashFiles('test_requirements.txt') }}
+        key: test-0-${{ env.pythonLocation }}-${{ env.os }}-${{ hashFiles('setup.cfg') }}-${{ hashFiles('test_requirements.txt') }}
```

Comment:
I think this was one real problem -- sharing the cache implicitly across Ubuntu versions (20.04 and 22.04)